types/prettier/index.d.ts

```diff
@@ -24,20 +29,160 @@ export type LiteralUnion<T extends U, U = string> = T | (Pick<U, never> & { _?:
 export type AST = any;
 export type Doc = doc.builders.Doc;
 
-// https://github.com/prettier/prettier/blob/main/src/common/ast-path.js
+// The type of elements that make up the given array T.
+type ArrayElement<T> = T extends Array<infer E> ? E : never;
```

Comment:
You can more easily get the element type of an array using `T[number]`

Reply:
Yes definitely, but I'm not sure how you would incorporate that into the `each` and `map` function definitions. The thing is `Path<T>` can have any type in the place of `T`. So if I want to type `each(callback: EachCallback<T>): void;`, I think I have to use it this way. Would love to know if there's another way though!